substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/api/SharedHostVM.java

```diff
@@ -0,0 +1,155 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.graal.pointsto.api;
+
+import com.oracle.graal.pointsto.BigBang;
+import com.oracle.graal.pointsto.PointsToAnalysis;
+import com.oracle.graal.pointsto.meta.AnalysisMethod;
+import com.oracle.graal.pointsto.meta.AnalysisType;
+import com.oracle.graal.pointsto.meta.AnalysisUniverse;
+import com.oracle.graal.pointsto.meta.HostedProviders;
+import jdk.vm.ci.meta.ResolvedJavaType;
+import org.graalvm.compiler.core.common.spi.ForeignCallDescriptor;
+import org.graalvm.compiler.core.common.spi.ForeignCallsProvider;
+import org.graalvm.compiler.java.GraphBuilderPhase;
+import org.graalvm.compiler.nodes.StructuredGraph;
+import org.graalvm.compiler.nodes.graphbuilderconf.GraphBuilderConfiguration;
+import org.graalvm.compiler.nodes.graphbuilderconf.IntrinsicContext;
+import org.graalvm.compiler.options.OptionValues;
+import org.graalvm.compiler.phases.OptimisticOptimizations;
+import org.graalvm.nativeimage.hosted.Feature;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ForkJoinPool;
+import java.util.function.BiConsumer;
+
+public abstract class SharedHostVM implements HostVM {
+    private final List<BiConsumer<Feature.DuringAnalysisAccess, Class<?>>> classReachabilityListeners;
+    private final OptionValues options;
+    protected final ClassLoader classLoader;
+
+    protected final List<BiConsumer<AnalysisMethod, StructuredGraph>> methodAfterParsingHooks = new CopyOnWriteArrayList<>();
+
+    public void addMethodAfterParsingHook(BiConsumer<AnalysisMethod, StructuredGraph> methodAfterParsingHook) {
+        methodAfterParsingHooks.add(methodAfterParsingHook);
+    }
+
+    protected SharedHostVM(OptionValues options, ClassLoader classLoader) {
+        this.options = options;
+        this.classLoader = classLoader;
+        this.classReachabilityListeners = new ArrayList<>();
+    }
+
+    @Override
+    public OptionValues options() {
+        return options;
+    }
+
+    @Override
+    public boolean isRelocatedPointer(Object originalObject) {
+        return false;
+    }
+
+    @Override
+    public void clearInThread() {
+
+    }
+
+    @Override
+    public void installInThread(Object vmConfig) {
+        Thread.currentThread().setContextClassLoader(classLoader);
+    }
+
+    @Override
+    public Object getConfiguration() {
```

Comment:
These empty implementations (`getConfiguration`, `checkForbidden`, `registerType`, `handleForeignCall`, `checkType`, `methodBeforeTypeFlowCreationHook`) can be moved in `HotSpotHost`. They are confusing here since they do have an implementation in `SVMHost`. Alternatively,  they can be made default empty in the `HostVM` interface.

Reply:
Fixed